From akinobu.mita@gmail.com Sun Feb 17 01:15:08 2008
Return-Path: <akinobu.mita@gmail.com>
Received: from imap.linux.ibm.com ([unix socket]) by imap.linux.ibm.com
	(Cyrus v2.3.7-Invoca-RPM-2.3.7-7) with LMTPA; Sun, 17 Feb 2008 01:15:08
	-0500
X-Sieve: CMU Sieve 2.3
Received: by imap.linux.ibm.com (Postfix, from userid 101) id 882DA191015F;
	Sun, 17 Feb 2008 01:15:08 -0500 (EST)
X-Spam-TestScore: SPF_NEUTRAL=1.379
X-Spam-TokenSummary: Bayes not run.
X-Spam-Checker-Version: SpamAssassin 3.1.7 (2006-10-05) on
	imap.linux.ibm.com
X-Spam-Level: *
X-Spam-Status: No, score=1.4 required=5.0 tests=SPF_NEUTRAL 
	autolearn=disabled version=3.1.7
X-Spam-Relay-Country: US ** US US US ** US US XX JP
Received: from smtp.linux.ibm.com (smtp.linux.ibm.com [9.26.4.197]) by
	imap.linux.ibm.com (Postfix) with ESMTP id DC875191009F for
	<cmm@imap.linux.ibm.com>; Sun, 17 Feb 2008 01:15:03 -0500 (EST)
Received: from localhost (localhost.localdomain [127.0.0.1]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id 5F770C03F for
	<cmm@linux.ibm.com>; Sun, 17 Feb 2008 01:15:03 -0500 (EST)
X-Virus-Scanned: amavisd-new at linux.ibm.com
Received: from VMSDVMA.POK.IBM.COM (vmsdvma.pok.ibm.com [9.56.231.65]) by
	smtp.linux.ibm.com (Postfix) with ESMTP id 10C38C03E for
	<cmm@linux.ibm.com>; Sun, 17 Feb 2008 01:15:02 -0500 (EST)
Received:  by VMSDVMA.POK.IBM.COM (IBM VM SMTP Level 520) via spool with
	SMTP id 0127 ; Sun, 17 Feb 2008 01:13:39 EST
Received: by vmsdvma.vnet.ibm.com (xagent2 5.9.5) via xagsmtp3 with spool
	id 8612 for cmm@linux.vnet.ibm.com; Sun, 17 Feb 2008 01:13:39 -0500 (EST)
Received: from d01relay06.pok.ibm.com [9.56.227.116] by VMSDVMA.POK.IBM.COM
	(IBM VM SMTP Level 520) via TCP with ESMTP ; Sun, 17 Feb 2008 01:13:39 EST
Received: from d01av05.pok.ibm.com (d01av05.pok.ibm.com [9.56.224.195]) by
	d01relay06.pok.ibm.com (8.13.8/8.13.8/NCO v8.7) with ESMTP id
	m1H6F2xM2404480 for <cmm@us.ibm.com>; Sun, 17 Feb 2008 01:15:02 -0500
Received: from d01av05.pok.ibm.com (loopback [127.0.0.1]) by
	d01av05.pok.ibm.com (8.12.11.20060308/8.13.3) with ESMTP id m1H6ERKh011135
	for <cmm@us.ibm.com>; Sun, 17 Feb 2008 01:14:28 -0500
Received: from d01as01.pok.ibm.com (d01as01 [9.56.224.36]) by
	d01av05.pok.ibm.com (8.12.11.20060308/8.12.11) with ESMTP id m1H6ERMu011130
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK) for
	<cmm@us.ibm.com>; Sun, 17 Feb 2008 01:14:27 -0500
Received: from e1.ny.us.ibm.com (e1.pok.ibm.com [9.56.232.141]) by
	d01as01.pok.ibm.com (8.12.11.20060308/8.12.11) with ESMTP id m1H6F0M2021005
	(version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-SHA bits=256 verify=OK) for
	<cmm@us.ibm.com>; Sun, 17 Feb 2008 01:15:00 -0500
Received: from wa-out-1112.google.com (wa-out-1112.google.com
	[209.85.146.179]) by e1.ny.us.ibm.com (8.13.8/8.13.8) with ESMTP id
	m1H6Exq2031244 for <cmm@us.ibm.com>; Sun, 17 Feb 2008 01:14:59 -0500
Received: by wa-out-1112.google.com with SMTP id k34so2310477wah.10 for
	<cmm@us.ibm.com>; Sat, 16 Feb 2008 22:14:58 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;        d=gmail.com;
	s=gamma;
	h=domainkey-signature:received:received:date:from:to:cc:subject:message-id:mime-version:content-type:content-disposition:user-agent;
	       bh=7DjV8ikrb9ph5Z9+hZEGYreY4W9G0kbsTS34S/YsKF0=;       
	b=vT6m39jKZpjLBC2giaVu0v7v8q9oX54z2CNzV+wAjP25GI9goBDDYUZeCdF+IaKjGZEcEPdg4S9r45BbNEOYKMY/hACtTL06KnpoIHe7Fu3UL3vNkYggvvbnGbRle0pej4hZfSA0paOD74o0bwA1e6Dl4+E5Qie4l4pVDz9n044=
DomainKey-Signature: a=rsa-sha1; c=nofws;        d=gmail.com; s=gamma;
	h=date:from:to:cc:subject:message-id:mime-version:content-type:content-disposition:user-agent;
	      
	b=DZXox896eziC0pJq01x88Dv5fETCsnTGa3BSMXZEfAhosl2/cUZiaw63aU2wbd3HQjazw++34x74P+vo1rJv37jqWhr2V1J+QJOUnMS4I5BdNEb/5mK58Ssa+zDbwp1c07Usfeilnmb+yH9IZXbny3lSoXdXvhXADcrxBOs+QI8=
Received: by 10.114.174.2 with SMTP id w2mr1972542wae.17.1203228897981;
	Sat, 16 Feb 2008 22:14:57 -0800 (PST)
Received: from @ ( [222.148.118.78]) by mx.google.com with ESMTPS id
	z15sm12804257pod.11.2008.02.16.22.14.55 (version=SSLv3 cipher=OTHER); Sat,
	16 Feb 2008 22:14:57 -0800 (PST)
Date: Sun, 17 Feb 2008 15:08:40 +0900
From: Akinobu Mita <akinobu.mita@gmail.com>
To: linux-ext4@vger.kernel.org
Cc: Andrew Morton <akpm@linux-foundation.org>,        Stephen Tweedie <sct@redhat.com>, adilger@clusterfs.com,        Mingming Cao <cmm@us.ibm.com>, Theodore Tso <tytso@MIT.EDU>
Subject: [PATCH 1/4] ext4: use ext4_group_first_block_no()
Message-ID: <20080217060839.GH3390@APFDCB5C>
MIME-Version: 1.0
Content-Type: text/plain; charset=iso-2022-jp
Content-Disposition: inline
User-Agent: Mutt/1.5.17 (2007-11-01)
X-Xagent-From: akinobu.mita@gmail.com
X-Xagent-To: cmm@linux.vnet.ibm.com
X-Xagent-Gateway: vmsdvma.vnet.ibm.com (XAGENTU at VMSDVMA)
X-Evolution-Source: imap://cmm@imap.linux.ibm.com/
Content-Transfer-Encoding: 8bit

Use ext4_group_first_block_no() and assign the return values to
ext2_fsblk_t variables.

Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
Cc: Stephen Tweedie <sct@redhat.com>
Cc: adilger@clusterfs.com
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: Mingming Cao <cmm@us.ibm.com>
Cc: Theodore Tso <tytso@MIT.EDU>
---
 fs/ext4/balloc.c |    6 +++---
 fs/ext4/xattr.c  |    6 ++----
 2 files changed, 5 insertions(+), 7 deletions(-)

Index: 2.6-rc/fs/ext4/xattr.c
===================================================================
--- 2.6-rc.orig/fs/ext4/xattr.c
+++ 2.6-rc/fs/ext4/xattr.c
@@ -808,10 +808,8 @@ inserted:
 			get_bh(new_bh);
 		} else {
 			/* We need to allocate a new block */
-			ext4_fsblk_t goal = le32_to_cpu(
-					EXT4_SB(sb)->s_es->s_first_data_block) +
-				(ext4_fsblk_t)EXT4_I(inode)->i_block_group *
-				EXT4_BLOCKS_PER_GROUP(sb);
+			ext4_fsblk_t goal = ext4_group_first_block_no(sb,
+						EXT4_I(inode)->i_block_group);
 			ext4_fsblk_t block = ext4_new_block(handle, inode,
 							goal, &error);
 			if (error)
Index: 2.6-rc/fs/ext4/balloc.c
===================================================================
--- 2.6-rc.orig/fs/ext4/balloc.c
+++ 2.6-rc/fs/ext4/balloc.c
@@ -48,7 +48,6 @@ void ext4_get_group_no_and_offset(struct
 unsigned ext4_init_block_bitmap(struct super_block *sb, struct buffer_head *bh,
 		 ext4_group_t block_group, struct ext4_group_desc *gdp)
 {
-	unsigned long start;
 	int bit, bit_max;
 	unsigned free_blocks, group_blocks;
 	struct ext4_sb_info *sbi = EXT4_SB(sb);
@@ -106,11 +105,12 @@ unsigned ext4_init_block_bitmap(struct s
 	free_blocks = group_blocks - bit_max;

 	if (bh) {
+		ext4_fsblk_t start;
+
 		for (bit = 0; bit < bit_max; bit++)
 			ext4_set_bit(bit, bh->b_data);

-		start = block_group * EXT4_BLOCKS_PER_GROUP(sb) +
-			le32_to_cpu(sbi->s_es->s_first_data_block);
+		start = ext4_group_first_block_no(sb, block_group);

 		/* Set bits for block and inode bitmaps, and inode table */
 		ext4_set_bit(ext4_block_bitmap(sb, gdp) - start, bh->b_data);
